slackbot.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"log"
+	"time"
+
+	pagerduty "github.com/PagerDuty/go-pagerduty"
+	"github.com/shomali11/slacker"
+)
+
+// Init TODO
+func Init(config *Config) {
+	log.Println("Connected!")
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var opts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, opts); err != nil {
+		log.Fatal(err)
+	} else {
+		log.Println("Configured schedule is \"" + schedule.Name + "\" with ID: " + config.PagerDuty.Schedule)
+	}
+}
+
+// Err TODO
+func Err(err string) {
+	log.Println(err)
+}
+
+// Emergency TODO
+func Emergency(request slacker.Request, response slacker.ResponseWriter, config *Config) {
+	client := pagerduty.NewClient(config.PagerDuty.APIKey)
+	var scheduleOpts pagerduty.GetScheduleOptions
+	if schedule, err := client.GetSchedule(config.PagerDuty.Schedule, scheduleOpts); err != nil {
+		log.Fatal(err)
+	} else {
+		log.Println("Opening incident on schedule \"" + schedule.Name +
+			"\"/" + config.PagerDuty.Schedule)
+	}
+
+	newIncident := pagerduty.CreateIncidentOptions{
+		Type: "incident",
+		Title: "Incident opened by " + config.CustomerName +
```

Comment:
Here and below, please use fmt.Spritnf formatting

Reply:
fixed.